data-prepper-core/src/main/java/org/opensearch/dataprepper/core/pipeline/server/DynamicPipelineUpdateUtil.java

```diff
@@ -0,0 +1,130 @@
+import org.opensearch.dataprepper.model.configuration.PipelineModel;
+import org.opensearch.dataprepper.model.configuration.PipelinesDataFlowModel;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+import org.opensearch.dataprepper.model.configuration.SinkModel;
+import org.opensearch.dataprepper.model.peerforwarder.RequiresPeerForwarding;
+import org.reflections.Reflections;
+import org.reflections.scanners.TypeAnnotationsScanner;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DynamicPipelineUpdateUtil {
+
+    public static final Set<String> excludedProcessorsFromDynamicUpdate = Set.of("aggregate");
+
+    public static boolean isDynamicUpdateFeasible(final PipelinesDataFlowModel currentPipelineDataFlowMode,
+                                                  final PipelinesDataFlowModel targetDataPipelineDataFlowMode) {
+
+        checkIfSamePipelineExists(currentPipelineDataFlowMode, targetDataPipelineDataFlowMode);
+        Set<String> singleThreadedProcessorNames = scanForSingleThreadAnnotatedProcessorPlugins();
+        singleThreadedProcessorNames.addAll(excludedProcessorsFromDynamicUpdate);
+
+        for (String pipelineName : currentPipelineDataFlowMode.getPipelines().keySet()) {
+            PipelineModel currentPipelineModel = currentPipelineDataFlowMode.getPipelines().get(pipelineName);
+            PipelineModel targetPipelineModel = targetDataPipelineDataFlowMode.getPipelines().get(pipelineName);
+
+            // Check if source configuration remains unchanged
+            PluginModel currentSource = currentPipelineModel.getSource();
+            PluginModel targetSource = targetPipelineModel.getSource();
+            if (!currentSource.equals(targetSource)) {
+                throw new DynamicPipelineConfigUpdateException(
+                        "Source configuration cannot be modified in pipeline: " + pipelineName);
+            }
+
+            // Check if sinks configuration remains unchanged
+            List<SinkModel> currentSinks = currentPipelineModel.getSinks();
+            List<SinkModel> targetSinks = targetPipelineModel.getSinks();
+            if (!currentSinks.equals(targetSinks)) {
+                throw new DynamicPipelineConfigUpdateException(
+                        "Sinks configuration cannot be modified in pipeline: " + pipelineName);
+            }
+
+            List<PluginModel> currentProcessors = currentPipelineModel.getProcessors();
+            List<PluginModel> targetProcessors = targetPipelineModel.getProcessors();
+
+            currentProcessors = currentProcessors == null ? List.of() : currentProcessors;
+            targetProcessors = targetProcessors == null ? List.of() : targetProcessors;
+
+            // Collect single-threaded processors in current and target
+            Set<String> currentSingleThreaded = currentProcessors.stream()
```

Comment:
There is a bit of split logic here. We have other places to look for this annotation. Can we make use of that existing code?